app/Models/Foundation/Summit/Events/Presentations/Presentation.php

```diff
@@ -1734,11 +1734,48 @@ public function setCustomOrder(int $custom_order): void
         $this->custom_order = $custom_order;
     }
 
+    /**
+     * @param $begin_voting_date
+     * @param $end_voting_date
+     * @return ArrayCollection
+     */
+    private function getVotesRange($begin_voting_date = null, $end_voting_date = null): ArrayCollection {
+        $criteria = null;
+
+        if ($begin_voting_date != null) {
+            $begin_voting_date = Carbon::createFromTimestamp($begin_voting_date);
+            $criteria = Criteria::create();
+            $criteria->where(Criteria::expr()->gte('created', $begin_voting_date));
+        }
+        if ($end_voting_date != null) {
+            $end_voting_date = Carbon::createFromTimestamp($end_voting_date);
+            $expr = Criteria::expr()->lte('created', $end_voting_date);
+            if ($criteria == null) {
+                $criteria = Criteria::create();
+                $criteria->where($expr);
+            } else {
+                $criteria->andWhere($expr);
+            }
+        }
+        $res = $criteria != null ? $this->attendees_votes->matching($criteria) : $this->attendees_votes;
+        return new ArrayCollection($res->toArray());
+    }
+
+    /**
+     * @return array|String[]
+     */
+    public function getVoters($begin_voting_date = null, $end_voting_date = null): array {
+        return $this->getVotesRange($begin_voting_date, $end_voting_date)->map(function ($attendeeVote) {
+            $voter = $attendeeVote->getVoter();
```

Comment:
dont do this
pass the voter attendee entity

Reply:
done as requested